### PR TITLE
ssl/ssl_ciph.c: Replace size_t with int and add the checks

### PR DESCRIPTION
**Pull request by JiangJias**
Replace the type of "mac", "out", and "blk" with int to avoid implicit conversion when it is assigned by EVP_MD_get_size(), EVP_CIPHER_get_iv_length(), and EVP_CIPHER_get_block_size(). Moreover, add the checks to avoid integer overflow.

Fixes: 045bd04706 ("Add DTLS_get_data_mtu() function")

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by t8m — CHANGES_REQUESTED** [review]

File: ssl/ssl_ciph.c [hunk 1]
```diff
@@ -2220,8 +2223,11 @@ int ssl_cipher_get_overhead(const SSL_CIPHER *c, size_t *mac_overhead,
 
             in = 1; /* padding length byte */
             out = EVP_CIPHER_get_iv_length(e_ciph);
+            if (out < 0) {
+                return 0;
+            }
             blk = EVP_CIPHER_get_block_size(e_ciph);
-            if (blk == 0)
+            if (blk <= 0)
```
[the remark this answers is not in the document]
Comment (JiangJias) [reply ssl/ssl_ciph.c:2230]: Thanks. I have alter the check and add explicit casts in my new commit.

**Review by t8m — APPROVED** [review]

**Review by Sashan — APPROVED** [review]

File: ssl/ssl_ciph.c [hunk 2]
```diff
@@ -2208,6 +2208,9 @@ int ssl_cipher_get_overhead(const SSL_CIPHER *c, size_t *mac_overhead,
             return 0;
 
         mac = EVP_MD_get_size(e_md);
+        if (mac <= 0) {
```
Comment (slontis) [inline ssl/ssl_ciph.c:2211]: remove {} on single lines

File: ssl/ssl_ciph.c [hunk 3]
```diff
@@ -2220,16 +2223,19 @@ int ssl_cipher_get_overhead(const SSL_CIPHER *c, size_t *mac_overhead,
 
             in = 1; /* padding length byte */
             out = EVP_CIPHER_get_iv_length(e_ciph);
+            if (out < 0) {
```
Comment (slontis) [inline ssl/ssl_ciph.c:2226]: and here also

Comment (JiangJias) [reply ssl/ssl_ciph.c]: Thanks. I have submited a new commit to fix both of them.

**Review by t8m — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by t8m** [comment]
Squashed and merged to the master branch. Thank you for the contribution.

**Pull request closed without merging**